compiler/rustc_resolve/src/macros.rs

```diff
@@ -877,3 +906,13 @@ impl<'a> Resolver<'a> {
         result
     }
 }
+
+/// Convert the given number into the corresponding ordinal
+fn ordinalize(v: usize) -> String {
+    match ((11..=13).contains(&(v % 100)), v % 10) {
+        (false, 1) => "1st".to_string(),
+        (false, 2) => "2nd".to_string(),
+        (false, 3) => "3rd".to_string(),
+        _ => format!("{v}th"),
+    }
+}
```

Comment:
I'm not sure if this is helpful. Can it just be `this case is never used` instead, since we highlight the arm already?

Reply:
Arms might look pretty similar to each other. Not all editors show the line number. IMO it's better to give two ways to find the thing the lint complains about.

Reply:
> Not all editors show the line number. IMO it's better to give two ways to find the thing the lint complains about.

If some editors don't show line numbers, they should at least highlight the span that had the lint. There are a lot of errors and lints that don't make sense if there is no context given (line number, column number, etc.), and this would be one of them.

Reply:
Side note : `ordinalize(21)` yields `1st` and not `21st`, which I'm not sure is desired.

Reply:
Btw which arm are you referring to as the `1st`? The one with index `0` or the one with index `1`?

Reply:
> I've noticed that ordinalize(21) yields 1st and not 21st, which I'm not sure is desired.

Good catch! Indeed it was not desired. I have fixed it. I've wrote some tests but they didn't run because they were disabled in Cargo.toml, now they are enabled.

> Btw which arm are you referring to as the 1st? The one with index 0 or the one with index 1?

The one with index 0, there is a `+1` where `ordinalize` is being called. You can do some bikeshed on this question I guess, but note that it can be changed in the future. Personally I'm fine with either way.

Its ordinal (as well as line/column numbers) is the only watertight way to specify a macro arm. There is no rule that says that matchers, or even entire arms, have to be different from each other. For example, this is allowed without a warning currently:

```Rust
macro_rules! t {
    () => { println!("Hi") };
    () => { println!("Hi") };
    () => { println!("Hi") };
}
```

Ideally, there would be unused warnings for the second and third arm.

In practice, I admit this is rarer but it occurs often that macro arms look very similar to each other and you have to squint to figure out the differences.